lib/streamlit/runtime/caching/hashing.py

```diff
@@ -291,6 +288,11 @@ def _to_bytes(self, obj: Any) -> bytes:
 
             self.update(h, obj.tobytes())
             return h.digest()
+        elif type_util.is_type(obj, "PIL.Image.Image"):
+            import numpy as np
+
+            np_array = np.frombuffer(obj.tobytes(), dtype="uint8")
+            return self.to_bytes(np_array)
```

Comment:
We convert PIL image to numpy array, and then only compute hash of NumPy array to reuse logic of sampling NumPy array in case of large numpy array, more than 1_000_000 elements. 

In the case of a relatively large RGBA image 500px X 500px we reach the limit 500 * 500 * 4. = 1_000_000

Reply:
Oh, I see! (Will you add a comment above saying "we don't just hash the results of `obj.tobytes()` because we want to use the sampling logic for numpy data")